src/features/tags/components/TagManager/index.tsx

```diff
@@ -1,45 +1,50 @@
-import { useRouter } from 'next/router';
 import { useState } from 'react';
 
 import GroupToggle from './components/GroupToggle';
+import useCreateTag from 'features/tags/hooks/useCreateTag';
+import { useMessages } from 'core/i18n';
+import { useNumericRouteParams } from 'core/hooks';
+import useTagGroups from 'features/tags/hooks/useTagGroups';
+import useTags from 'features/tags/hooks/useTags';
 import { ZetkinTag } from 'utils/types/zetkin';
-import ZUIQuery from 'zui/ZUIQuery';
+import ZUIFutures from 'zui/ZUIFutures';
 import ZUISection from 'zui/ZUISection';
-import { tagGroupsResource, tagsResource } from 'features/tags/api/tags';
 import TagManagerController, {
   TagManagerControllerProps,
 } from './TagManagerController';
-import { useCreateTag, useEditTag } from './utils';
 
 import messageIds from '../../l10n/messageIds';
-import { useMessages } from 'core/i18n';
+import useEditTag from 'features/tags/hooks/useEditTag';
 
 type TagManagerProps = Omit<
   TagManagerControllerProps,
   'availableGroups' | 'availableTags' | 'onCreateTag' | 'onEditTag'
 > & { disableEditTags?: boolean; onTagEdited?: (tag: ZetkinTag) => void };
 
 const TagManager: React.FunctionComponent<TagManagerProps> = (props) => {
-  const { orgId } = useRouter().query;
-
-  const tagsQuery = tagsResource(orgId as string).useQuery();
-  const tagGroupsQuery = tagGroupsResource(orgId as string).useQuery();
+  const { orgId } = useNumericRouteParams();
+  const tagsFuture = useTags(orgId);
+  const { tagGroupsFuture } = useTagGroups(orgId);
 
-  const createTag = useCreateTag();
-  const editTag = useEditTag(props.onTagEdited);
+  const createTag = useCreateTag(orgId);
+  const editTag = useEditTag(orgId, props.onTagEdited);
```

Comment:
This is a new pattern that I haven't seen anywhere else. I would expect this to look like so:

```ts
const { updateTag } = useTagMutations(orgId);

const editTag = async (id, data) => {
  // Invoke updateTag() and wait for it to finish
  const tag = await updateTag(id, data);

  // Once it finished, the tag has been edited, so we can call the callback
  onTagEdited(tag);
};
```

The above is an example of the exercise "write the code you wish you had". Of course you will need to refactor `useEditTag` and `updateTag` to work the way the code above assumes. I'm trying to illustrate/invent a pattern, and doing so by writing the code you wish you had can be a good trick because it allows you to define the external interface of the hook first, and then go and make sure it works that way.